Source/WebCore/Modules/WebGPU/GPUAdapter.cpp

```diff
@@ -72,8 +73,7 @@ void GPUAdapter::requestDevice(ScriptExecutionContext&, const std::optional<GPUD
 
 void GPUAdapter::requestAdapterInfo(const std::optional<Vector<String>>&, RequestAdapterInfoPromise&& promise)
 {
-    // FIXME: https://bugs.webkit.org/show_bug.cgi?id=251377 - [WebGPU] Implement GPUAdapter.requestAdapterInfo
-    promise.resolve(nullptr);
+    promise.resolve(GPUAdapterInfo::create());
```

Comment:
I'm not sure this is sufficient - we will probably want this to be more complicated in the future.

I think we should have device creation automatically report this info from the GPU Process to the web process - like how limits and features are already handled

Reply:
I think we may have to or want to return from requestAdapterInfo prior to creating a device since the specification does not require a device to be created prior to `requestAdapterInfo` being called. The result of requestAdapterInfo may impact the options / features passed to `GPUAdapter.requestDevice`, it is foreseeable a website waits for the promise to resolve prior to creating a device.

I do agree that we may need to expand upon this in the future but it depends how the result of `requestAdapterInfo` is interpreted. The specification states that all the four strings can be empty:
https://gpuweb.github.io/gpuweb/#dom-gpuadapterinfo-vendor
https://gpuweb.github.io/gpuweb/#dom-gpuadapterinfo-architecture
https://gpuweb.github.io/gpuweb/#dom-gpuadapterinfo-device
https://gpuweb.github.io/gpuweb/#dom-gpuadapterinfo-description

WebGPUAdapter already has an adapter name, we could likely use this directly to populate at least one of the GPUAdapterInfo fields